__tests__/api/controllers/shoppingItem.controller.spec.ts

```diff
@@ -0,0 +1,171 @@
+import * as request from 'supertest';
+import App from '../../../src/app';
+import ShoppingItem from '../../../src/entity/ShoppingItem';
+import ShoppingItemDataSetup from '../../setup/shoppingItemData.setup';
+jest.mock('../../../src/api/middlewares/session.middleware');
+
+describe('ShoppingItems controller', () => {
+  let app: any;
+  let express: any;
+  const shoppingItemDataSetup = new ShoppingItemDataSetup();
+  const shoppingItemData = {
+    boughtById: 0,
+    boughtForId: 0,
+    categoryId: 0,
+    date: '2019-09-03 18:00:00',
+    name: 'Chleb',
+    price: 2.0,
+    quantity: 1.0,
+    shared: false,
+    shopId: 0,
+    unitDiscount: 0.5,
+  };
+  beforeAll(async () => {
+    app = new App({
+      logging: ['info'],
+      port: 0,
+    });
+    express = await app.run();
+  });
+
+  beforeEach(async () => {
+    for (const shoppingItem of await ShoppingItem.find()) {
+      await shoppingItem.remove();
+    }
+    await shoppingItemDataSetup.destroy();
+    await shoppingItemDataSetup.create();
+  });
+
+  afterAll(async () => {
+    await app.destroy();
+  });
+  function doCreateItem() {
+    shoppingItemData.boughtForId = shoppingItemDataSetup.user.id;
+    shoppingItemData.shopId = shoppingItemDataSetup.shop.id;
+    shoppingItemData.categoryId = shoppingItemDataSetup.category.id;
+    shoppingItemData.boughtById = shoppingItemDataSetup.user.id;
+  }
+
+  it('should get shoppingItems', async () => {
+    const response = await request(express)
+      .get('/shoppingItems')
+      .expect(200);
+    expect(response.body).toHaveLength(0);
+  });
+  it('should find an item', async () => {
+    doCreateItem();
+    let response = await request(express)
+      .post('/shoppingItems/')
+      .send(shoppingItemData)
+      .expect(201);
+    expect(response.body.name).toEqual(shoppingItemData.name);
+    response = await request(express)
+      .get('/shoppingItems/' + response.body.id)
+      .expect(200);
+    expect(response.body.name).toEqual(shoppingItemData.name);
+  });
+  it('should find and fail when shoppingItem does not exist', async () => {
+    await request(express)
+      .get('/shoppingItems/100')
+      .expect(404);
+  });
+  it('should create shoppingItem', async () => {
+    doCreateItem();
+    const response = await request(express)
+      .post('/shoppingItems/')
+      .send(shoppingItemData)
+      .expect(201);
+    expect(response.body.price).toEqual(shoppingItemData.price);
+    expect(response.body.shop.id).toEqual(shoppingItemData.shopId);
+    expect(response.body.id).toBeGreaterThan(0);
+    expect(await ShoppingItem.findOne(response.body.id)).not.toBeUndefined();
+  });
+  it('should update primitive value of ShoppingItem', async () => {
+    doCreateItem();
+    let response = await request(express)
```

Comment:
In `update` test there is no need to create item using `POST` request. Use model instead. 
You can wrap it in method `doCreateItem` - this time named correctly.